src/views/DidUpgradePromo/DidUpgradePromo.tsx

```diff
@@ -0,0 +1,119 @@
+import { useCallback, useState } from 'react';
+import { Link, useParams } from 'react-router-dom';
+import { browser } from 'webextension-polyfill-ts';
+
+import * as styles from './DidUpgradePromo.module.css';
+
+import { Identity } from '../../utilities/identities/types';
+import {
+  PasswordField,
+  usePasswordField,
+} from '../../components/PasswordField/PasswordField';
+import { saveIdentity } from '../../utilities/identities/identities';
+import { IdentitySlide } from '../../components/IdentitySlide/IdentitySlide';
+import { TxStatusModal } from '../../components/TxStatusModal/TxStatusModal';
+import { LinkBack } from '../../components/LinkBack/LinkBack';
+import { Stats } from '../../components/Stats/Stats';
+import { paths } from '../paths';
+import {
+  createDid,
+  getDidCreationDetails,
+  waitFinalised,
+} from '../../utilities/didUpgradePromo/didUpgradePromo';
+
+interface Props {
+  identity: Identity;
+}
+
+export function DidUpgradePromo({ identity }: Props): JSX.Element | null {
+  const t = browser.i18n.getMessage;
+
+  const [txHash, setTxHash] = useState<string>();
+
+  const [submitting, setSubmitting] = useState(false);
+  const [status, setStatus] = useState<'pending' | 'success' | 'error' | null>(
+    null,
+  );
+
+  const { submitter } = useParams() as { submitter: string };
+
+  const passwordField = usePasswordField();
+
+  const handleSubmit = useCallback(
+    async (event) => {
+      event.preventDefault();
+
+      try {
+        const { seed } = await passwordField.get(event);
+
+        setSubmitting(true);
+        setStatus('pending');
+
+        const creationDetails = await getDidCreationDetails(seed, submitter);
+        console.log('creation details: ', creationDetails);
+
+        const { tx_hash } = await createDid(creationDetails);
+        console.log('tx hash: ', tx_hash);
```

Comment:
console.logs should leave

Reply:
Why am I still forgetting to remove these! I'm embarrassed.